go/vt/sidecardb/sidecardb.go

```diff
@@ -202,6 +208,31 @@ func Init(ctx context.Context, exec Exec) error {
 	return nil
 }
 
+// We need to allow zero dates for existing sidecar database tables which may happen to
+// actually have zero (in) date values. setSQLMode gets the current sql_mode, change it to a more relaxed value,
+// defer restoring it to the original value.
+func (si *schemaInit) setSQLMode() (func(), error) {
```

Comment:
I think this should be `setPermissiveSQLMode()` that effectively removes all restrictions.

Reply:
fixed